include/dlaf/matrix/copy.h

```diff
@@ -0,0 +1,43 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <hpx/hpx.hpp>
+
+#include "dlaf/matrix/copy_tile.h"
+#include "dlaf/types.h"
+
+namespace dlaf {
+
+/// Copy values from another matrix
+///
+/// Given a matrix with the same geometries and distribution, this function submits tasks that will
+/// perform the copy of each tile
+template <
+    template <class, Device> class MatrixTypeSrc, template <class, Device> class MatrixTypeDst,
+    class Tsrc, class Tdst, Device device_source, Device device_dest,
+    std::enable_if_t<std::is_same<const Tsrc, const Tdst>::value && !std::is_const<Tdst>::value, int> = 0>
```

Comment:
Since things might change regarding how `Device` is handled, I suggest we assume that both matrices have the same `Device` type to reduce complexity in the meantime. The `lacpy` function defined earlier assumes the `Scalar` type (`T`) to be equal, we should either generalize lacpy for Tiles to handle different types or assume that (`Tsrc = Tdst`).

Reply:
About `Device`, it's ok for me, we can extend it later.

About `Tsrc==Tdst`, I'd like to highlight the constraint specified through the `std::enable_if` directive:
- `Tsrc` and `Tdst` must be the same type, except for constness
- `Tdst` cannot be a const type
This allows to use any tile, compatible with the purpose, without having to force the conversion between const/non-const (I'm talking about the pointed type).

If you have any suggestion about this, I'll be glad to take them into account.

Reply:
I was wrong about this, as we discussed offline you are right about it.

I forgot that I can do `ConstMatrix& = Matrix&` (the problem exists with other types).

Thank you for pointing it out, I'm going to fix that!